### PR TITLE
Fix issue 206

### PR DESCRIPTION
**Pull request by saledouble**
Related to #206 

A screenshot of testing result:

image

### CODE REVIEW

**Comment by saledouble** [comment]
Default checking should be cleared. Blocked by #204

File: edu.cuny.hunter.log.evalution/src/edu/cuny/hunter/log/evaluation/handlers/EvaluationHandler.java [hunk 1]
```diff
@@ -59,13 +59,15 @@
 	private static final String EVALUATION_PROPERTIES_FILE_NAME = "eval.properties";
 	private static final String NOT_LOWER_LOG_LEVEL_IF_STATEMENT_KEY = "edu.cuny.hunter.log.evaluation.notLowerLogLevelInIfStatement";
 	private static final String NOT_LOWER_LOG_LEVEL_CATCH_BLOCK_KEY = "edu.cuny.hunter.log.evaluation.notLowerLogLevelInCatchBlock";
+	private static final String NOT_LOWER_LOG_LEVEL_KEYWORDS_KEY = "edu.cuny.hunter.log.evaluation.notLowerLogLevelWithKeywords";
```
Comment (khatchad) [inline edu.cuny.hunter.log.evalution/src/edu/cuny/hunter/log/evaluation/handlers/EvaluationHandler.java]: Is this documented in the wiki?

File: edu.cuny.hunter.log.ui/src/edu/cuny/hunter/log/ui/wizards/LogWizard.java [hunk 2]
```diff
@@ -142,9 +144,16 @@ public void createControl(Composite parent) {
 
 			// set up buttons.
 			Button checkButton4 = this.addBooleanButton(
+					"Never lower the logging level of logging statements with particular keywords in their messages.",
+					NOT_LOWER_LOG_LEVEL_KEY_WORDS, this.getProcessor()::setNotLowerLogLevelWithKeyWords, result,
+					SWT.CHECK);
+			checkButton4.setSelection(true);
```
Comment (khatchad) [inline edu.cuny.hunter.log.ui/src/edu/cuny/hunter/log/ui/wizards/LogWizard.java]: These should be read from a store. That's the problem with the other bug. https://github.com/ponder-lab/Optimize-Java-8-Streams-Refactoring/blob/a648cce999a36c698aef520e9d8ce516cd50301e/edu.cuny.hunter.streamrefactoring.ui/src/edu/cuny/hunter/streamrefactoring/ui/wizards/OptimizeStreamRefactoringWizard.java#L74

File: edu.cuny.hunter.log.evalution/src/edu/cuny/hunter/log/evaluation/handlers/EvaluationHandler.java (hunk 1 above)
Comment (saledouble) [reply edu.cuny.hunter.log.evalution/src/edu/cuny/hunter/log/evaluation/handlers/EvaluationHandler.java]: Yes. https://github.com/ponder-lab/Logging-Level-Evolution-Plugin/wiki/Running-the-Evaluator

File: edu.cuny.hunter.log.core/src/edu/cuny/hunter/log/core/analysis/LogAnalyzer.java [hunk 3]
```diff
@@ -60,6 +65,12 @@
 
 	private LinkedList<Float> boundary;
 
+	/**
+	 * A set of keywords in log messages.
+	 */
+	final private Set<String> keyWordsInLogMessages = Stream.of("failed", "disabled", "error", "exception")
```
Comment (khatchad) [inline edu.cuny.hunter.log.core/src/edu/cuny/hunter/log/core/analysis/LogAnalyzer.java]: Change "failed" to "fail" and "disabled" to "disable"

File: edu.cuny.hunter.log.core/src/edu/cuny/hunter/log/core/analysis/LogAnalyzer.java [hunk 4]
```diff
@@ -60,6 +65,12 @@
 
 	private LinkedList<Float> boundary;
 
+	/**
+	 * A set of keywords in log messages.
+	 */
+	private final Set<String> KEYWORDS_IN_LOG_MESSAGES = Stream.of("fail", "disable", "error", "exception")
```
Comment (khatchad) [inline edu.cuny.hunter.log.core/src/edu/cuny/hunter/log/core/analysis/LogAnalyzer.java]: Add "collision", "reboot", "terminate", "throw", "shut", "start", "shut", and "run".

**Pull request merged**